src/Maker/MakeAuthenticator.php

```diff
@@ -107,14 +166,62 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
                     $this->fileManager->getFileContents($path),
                     $input->getOption('firewall-name'),
                     $input->getOption('entry-point'),
-                    $classNameDetails->getFullName()
+                    $input->getArgument('authenticator-class')
                 );
                 $generator->dumpFile($path, $newYaml);
                 $securityYamlUpdated = true;
             } catch (YamlManipulationFailedException $e) {
             }
         }
 
+        if (self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type')) {
```

Comment:
Let's put the body of all of this into a private function for readability - e.g. `$this->generateLoginForm(...);`